File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/commands/low_level/outtake/ToggleOuttakeClawCommand.java
package org.firstinspires.ftc.teamcode.commands.low_level.outtake;

import com.arcrobotics.ftclib.command.CommandBase;
import com.arcrobotics.ftclib.command.CommandScheduler;

import org.firstinspires.ftc.teamcode.managers.OuttakeManager;
import org.firstinspires.ftc.teamcode.util.GamepadPlus;

public class ToggleOuttakeClawCommand extends CommandBase {

    OuttakeManager outtakeManager;
    GamepadPlus gamepad_codriver;
    public ToggleOuttakeClawCommand(OuttakeManager outtakeManager, GamepadPlus gamepad_codriver) {
        this.outtakeManager = outtakeManager;
        this.gamepad_codriver = gamepad_codriver;
    }

    @Override
    public void initialize(){
        if (!outtakeManager.isClosed()){
            gamepad_codriver.rumble(200);
            CommandScheduler.getInstance().schedule(
                    new SetOuttakeClawStateCommand(outtakeManager, OuttakeManager._OuttakeClawServoState.CLOSED)
            );
        }
        else {
            CommandScheduler.getInstance().schedule(
                    new SetOuttakeClawStateCommand(outtakeManager, OuttakeManager._OuttakeClawServoState.OPEN)
            );
        }
    }

    @Override
    public boolean isFinished() {
        return true;
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/commands/low_level/outtake/ToggleOuttakeClawCommand.kt
package org.firstinspires.ftc.teamcode.commands.low_level.outtake

import com.arcrobotics.ftclib.command.CommandBase
import com.arcrobotics.ftclib.command.CommandScheduler
import org.firstinspires.ftc.teamcode.managers.OuttakeManager
import org.firstinspires.ftc.teamcode.util.GamepadPlus

/**
 *
 * @property outtakeManager OuttakeManager
 * @property gamepadCodriver GamepadPlus
 * @property gamepadDriver GamepadPlus
 * @constructor
 */
class ToggleOuttakeClawCommand(
    private val outtakeManager: OuttakeManager,
    private val gamepadCodriver: GamepadPlus,
    private val gamepadDriver: GamepadPlus
) : CommandBase() {

    override fun initialize() {
        if (!outtakeManager.GetClawState()) {

            // Notify driver tp boost forward
            gamepadDriver.rumble(200)

            CommandScheduler.getInstance().schedule(
                SetOuttakeClawStateCommand(
                    outtakeManager,
                    OuttakeManager._OuttakeClawServoState.GRIP
                )
            )
        } else {
            CommandScheduler.getInstance().schedule(
                SetOuttakeClawStateCommand(
                    outtakeManager,
                    OuttakeManager._OuttakeClawServoState.RELEASE
                )
            )
        }
    }

    override fun isFinished(): Boolean {
        return true
    }
}
